src/util.c

```diff
@@ -775,15 +775,91 @@ void hyper_shutdown()
 	reboot(LINUX_REBOOT_CMD_POWER_OFF);
 }
 
-int hyper_cmd(char *cmd)
+/**
+ * Reads command stdout
+ *
+ * \param stdout_pipe command stdout fd
+ * \param[out] output command stdout
+ * \param[out] size command stdout size, if is NULL then it is not used
+ *
+ * \return 0 on success, else -1
+ */
+static int hyper_read_cmd_stdout(int stdout_pipe, uint8_t **output, size_t *size)
+{
+	size_t total_bytes_read = 0;
+	int bytes_read = 0;
+	size_t bufsize = 512;
+	uint8_t *buffer = NULL;
+
+	buffer = calloc(bufsize, sizeof(*buffer));
+
+	do {
+		bytes_read = read(stdout_pipe,
```

Comment:
The tests in this block do seem safe, but I'd still suggest checking if `bytes_read < 0` and handling that immediately rather than implicitly.